### PR TITLE
haste-client 0.2.2 (new formula)

### PR DESCRIPTION
**Pull request by jamebus**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/haste-client.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class HasteClient < Formula
+  desc "CLI client for haste-server"
+  homepage "https://hastebin.com/"
+  url "https://github.com/seejohnrun/haste-client.git",
+      :tag => "v0.2.2",
```
Comment (ilovezfs) [inline Formula/haste-client.rb]: Can we use https://github.com/seejohnrun/haste-client/archive/v0.2.2.tar.gz?

File: Formula/haste-client.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  def install
+    ENV["GEM_HOME"] = libexec
+    system "gem", "build", "haste.gemspec"
+    system "gem", "install", "haste-#{version}.gem"
+    bin.install libexec/"bin/haste"
+    bin.env_script_all_files(libexec/"bin", :GEM_HOME => ENV["GEM_HOME"])
+  end
+
+  test do
+    output = shell_output("echo testing | \"#{bin}/haste\"")
```
Comment (ilovezfs) [inline Formula/haste-client.rb]:
```
output = pipe_output("#{bin}/haste", "testing", 0)
```

**Review by ilovezfs — DISMISSED** [review]

**Comment by jamebus** [comment]
Yes, thank you. That's much better.

**Comment by ilovezfs** [comment]
@jamebus we'll also need to pin the dependencies:
```
diff --git a/Formula/haste-client.rb b/Formula/haste-client.rb
index c54c665..2449222 100644
--- a/Formula/haste-client.rb
+++ b/Formula/haste-client.rb
@@ -7,10 +7,25 @@ class HasteClient < Formula
 
   depends_on :ruby => "2.3"
 
+  resource "faraday" do
+    url "https://rubygems.org/gems/faraday-0.12.1.gem"
+    sha256 "0350b3d3adc9418e4e761198866709fc1b27db116cf52484989aa94d084a7df2"
+  end
+
+  resource "multipart-post" do
+    url "https://rubygems.org/gems/multipart-post-2.0.0.gem"
+    sha256 "3dc44e50d3df3d42da2b86272c568fd7b75c928d8af3cc5f9834e2e5d9586026"
+  end
+
   def install
     ENV["GEM_HOME"] = libexec
+    resources.each do |r|
+      r.verify_download_integrity(r.fetch)
+      system "gem", "install", r.cached_download, "--no-document",
+                    "--install-dir", libexec
+    end
     system "gem", "build", "haste.gemspec"
-    system "gem", "install", "haste-#{version}.gem"
+    system "gem", "install", "--ignore-dependencies", "haste-#{version}.gem"
     bin.install libexec/"bin/haste"
     bin.env_script_all_files(libexec/"bin", :GEM_HOME => ENV["GEM_HOME"])
   end
```

**Comment by jamebus** [comment]
Great idea! I've fixed that.

**Review by ilovezfs — APPROVED** [review]
```
haste-client:
  * C: 23: col 7: Use 2 (not 0) spaces for indentation.
  * W: 26: col 7: `end` at 26, 6 is not aligned with `resources.each do |r|` at 22, 4.
```
But otherwise :+1:

**Comment by ilovezfs** [comment]
@jamebus Thanks for your first contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**